### PR TITLE
[gbc] Handle forward- and backslashes in paths

### PR DESCRIPTION
**Pull request by chwarr**
Since .bond files can be authored on platforms with different directory
separator characters, the paths in build scripts and Bond `import`
statements may use a forwardslash when a backslash is expected or vice
versa.

In order to handle this, gbc now normalizes anything that is a
path (file paths, response file items, import directories, import
statement paths, output directories, &c.) to use the platform's
preferred directory separator character.

Tests for C++, C#, and Java have been updated to use some imports with
mixed slashes.

gbc doesn't use a dedicate type for file paths. `FilePath` is just a
type synonym for `String`, so some places may have been inadvertently
missed.

This does now mean that gbc cannot process files, on, say, Linux, with
backslashes in their names. This is expected to be rare, and is deemed
an acceptable compromise for a cross-platform tool.

Fixes https://github.com/Microsoft/bond/issues/869

### CODE REVIEW

**Comment by chwarr** [comment]
The clang build is broken. For the `#include` generated for `import` statements, we should force the slashes to always be forward.

**Comment by chwarr** [comment]
I think I've resolved the Clang build errors in commit a8f41fbc.

File: CHANGELOG.md [hunk 1]
```diff
@@ -42,7 +42,10 @@ different versioning scheme, following the Haskell community's
 * C++ codegen now applies the `--export-attribute` to the `ToString`,
   `FromString`, `ToEnum` and `FromEnum` functions.
 * `import` statements can now end with an optional semicolon.
-
+* File and directory paths on the command line, in response files, or in
```
Comment (lalo) [inline CHANGELOG.md]: what are response files?

Comment (chwarr) [reply CHANGELOG.md]: gbc response files are inputs that start with "@". If the input file is prefixed with "@", then gbc will treat each line of that file as an input file.

**Comment by chwarr** [comment]
@lalo: can you merge this?

**Comment by chwarr** [comment]
I just rebased this atop the recent changes to codegen and the C# project changes.

**Comment by lalo** [comment]
I'll merge this after CI.

**Comment by ara-ayvazyan** [comment]
Generated test files need to be regenerated.

**Comment by chwarr** [comment]
Thanks. I've fixed the generated test files.

**Pull request merged**